packages/trpc/react/trpc.ts

```diff
@@ -41,17 +69,45 @@ export const trpc = createTRPCNext<AppRouter, NextPageContext, "ExperimentalSusp
         }),
         splitLink({
           // check for context property `skipBatch`
-          condition: (op) => {
-            return op.context.skipBatch === true;
-          },
+          condition: (op) => !!op.context.skipBatch,
           // when condition is true, use normal request
-          true: httpLink({ url }),
-          // when condition is false, use batching
-          false: httpBatchLink({
-            url,
-            /** @link https://github.com/trpc/trpc/issues/2008 */
-            // maxBatchSize: 7
-          }),
+          true: (runtime) => {
+            const links = Object.fromEntries(
+              ENDPOINTS.map((endpoint) => [endpoint, httpLink({ url: url + "/" + endpoint })(runtime)])
+            );
+            return (ctx) => {
+              const parts = ctx.op.path.split(".");
+              let endpoint = '';
+              let path = '';
+              if (parts.length == 2) {
```

Comment:
This logic is gross. We should refactor the structure of the viewer, loggedInViewer and public routers to make this cleaner.